record/record.go

```diff
@@ -71,8 +71,6 @@
 // next block and looks for the next full or first chunk.
 package record // import "github.com/golang/leveldb/record"
 
-// TODO: implement the recovery algorithm.
```

Comment:
You're adding a bunch of new API to support implementing the recovery algorithm at a higher level, but remind me what the actual recovery algorithm is? There might be a more Go-like way to do this, but I can't remember what the 'this' is.